Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/BaseAutonomous.java

```diff
@@ -1,4 +1,240 @@
+                    telemetry.addData("- Position (Row/Col)", "%.0f / %.0f", row, col);
+                    telemetry.addData("- Size (Width/Height)", "%.0f / %.0f", width, height);
+                }
+                telemetry.update();
+            }
+        }
+        return position;
+    }
+
+    /**
+     * Moves the robot forward, backwards, and strafes right and left in inches.
+     * @param x strafes the robot right x inches. If moving left, pass in negative value.
+     * @param y moves the robot forward y inches. If moving backward, pass in negative value.
+     */
+    public void driveInches(double x, double y) {
+        double xTicks = x * TICKS_PER_INCH;
+        double yTicks = y * TICKS_PER_INCH;
+
+        double targetFL = xTicks + yTicks;
+        double targetFR = yTicks - xTicks;
+        double targetBL = yTicks - xTicks;
+        double targetBR = yTicks + xTicks;
+
+        // Determine new target position, and pass to motor controller
+        targetFL += motorFL.getCurrentPosition();
+        targetFR += motorFR.getCurrentPosition();
+        targetBL += motorBL.getCurrentPosition();
+        targetBR += motorBR.getCurrentPosition();
+
+        motorFL.setTargetPosition((int) targetFL);
+        motorFR.setTargetPosition((int) targetFR);
+        motorBL.setTargetPosition((int) targetBL);
+        motorBR.setTargetPosition((int) targetBR);
+
+        // Turn On RUN_TO_POSITION
+        motorFL.setMode(DcMotor.RunMode.RUN_TO_POSITION);
+        motorFR.setMode(DcMotor.RunMode.RUN_TO_POSITION);
+        motorBL.setMode(DcMotor.RunMode.RUN_TO_POSITION);
+        motorBR.setMode(DcMotor.RunMode.RUN_TO_POSITION);
+
+        // reset the timeout time and start motion.
+        runtime.reset();
+        motorFL.setPower(0.5);
+        motorFR.setPower(0.5);
+        motorBL.setPower(0.5);
+        motorBR.setPower(0.5);
+        // keep looping while we are still active, and there is time left, and both motors are running.
+        // Note: We use (isBusy() && isBusy()) in the loop test, which means that when EITHER motor hits
+        // its target position, the motion will stop.  This is "safer" in the event that the robot will
+        // always end the motion as soon as possible.
+        // However, if you require that BOTH motors have finished their moves before the robot continues
+        // onto the next step, use (isBusy() || isBusy()) in the loop test.
+        while (opModeIsActive() &&
+                (runtime.seconds() < 30) &&
+                (motorFL.isBusy() && motorFR.isBusy() && motorBL.isBusy() && motorBR.isBusy())) {
+        }
+
+        // Stop all motion;
+        stopDriving();
+
+        // Turn off RUN_TO_POSITION
+        motorFL.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        motorFR.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        motorBL.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        motorBR.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+    }
+    
+    // using imu
```

Comment:
Please use a javadoc comment for methods https://www.tutorialspoint.com/java/java_documentation.htm

Reply:
You didn't do this.